docs/pages/deploy-a-cluster/helm-deployments/migration-v12.mdx

```diff
@@ -0,0 +1,247 @@
+---
+title: Migrating to teleport-cluster v12
+description: How to upgrade to teleport-cluster version 12.
+---
+
+This guide covers the major changes of the `teleport-cluster` v12 chart
+and how to upgrade existing releases from version 11 to version 12.
+
+## Changes summary
+
+The main changes brought by the version 12 of the `teleport-cluster` chart are:
+
+- Teleport is now deployed in two components: auth and proxies.
+  Running Teleport with this new topology allows it to be more resilient to disruptions and scale better.
+- Proxies are now deployed as stateless workload. The `proxy` session recording mode uploads recordings asynchronously.
+  Non-uploaded records might be lost during rollouts (config changes or version upgrades for example).
+  `proxy-sync` ensures consistency and does not have this limitation.
+- `custom` mode is removed as it was broken by the topology change.
+  It is replaced by a new configuration override mechanism allowing you to pass arbitrary Teleport configuration.
+- the value `standalone.storage` that was previously deprecated in favor of `persistence` was removed.
+- The chart can now be scaled up in `standalone` mode. Proxy replication requires
+  a TLS certificate; Auth replication require using [HA storage backends](../../reference/backends.mdx).
+- The chart has always been versioned with Teleport but was often compatible with the previous Teleport major version.
+  This is not the case for v12. Using the chart v12 requires at least Teleport v12.
```

Comment:
I think this would be too constraining. e.g. after the v12 cut, the chart will be promoted to v13.0.0-dev and according to this criteria, it would not be able to run the v12 images.

I can add at least a warning in NOTES.txt.